tutorials/dataframe/df102_NanoAODDimuonAnalysis.C

```diff
@@ -30,84 +30,58 @@ void df102_NanoAODDimuonAnalysis()
+      p1.SetPtEtaPhiM(pt[0], eta[0], phi[0], mass[0]);
+      p2.SetPtEtaPhiM(pt[1], eta[1], phi[1], mass[1]);
 
-      TLorentzVector p2;
-      const auto i2 = idx[1];
-      p2.SetPtEtaPhiM(pt[i2], eta[i2], phi[i2], mass[i2]);
-
-      // Add four-vectors to build di-muon system and return the invariant mass
+      // Add four-vectors to build dimuon system and return the invariant mass
       return (p1 + p2).M();
    };
-   auto df_mass = df_pair.Define("Dimuon_mass", compute_mass,
-                                 {"Muon_pt", "Muon_eta", "Muon_phi", "Muon_mass", "Muon_pair"});
+   auto df_mass = df_os.Define("Dimuon_mass", compute_mass, {"Muon_pt", "Muon_eta", "Muon_phi", "Muon_mass"});
+
+   // Make histogram of dimuon mass spectrum
+   auto h = df_mass.Histo1D({"Dimuon_mass", "Dimuon_mass", 30000, 0.25, 300}, "Dimuon_mass");
 
-   // Produce histogram of di-muon mass spectrum
-   auto h = df_mass.Histo1D({"Dimuon_mass", "Dimuon_mass", 20000, 0.25, 300}, "Dimuon_mass")
-                   .GetValue();
+   // Request cut-flow report
+   auto report = df_mass.Report();
 
-   // Make plot
-   gStyle->SetOptStat(0);
-   gStyle->SetTextFont(42);
-   auto c = new TCanvas("c", "c", 800, 600);
-   c->SetLogx();
-   c->SetLogy();
+   // Produce plot
+   gStyle->SetOptStat(0); gStyle->SetTextFont(42);
+   auto c = new TCanvas("c", "", 800, 700);
+   c->SetLogx(); c->SetLogy();
 
-   h.SetTitle("");
-   h.GetXaxis()->SetTitle("Invariant di-muon mass (GeV)");
-   h.GetXaxis()->SetTitleSize(0.04);
-   h.GetYaxis()->SetTitle("N_{Events}");
-   h.GetYaxis()->SetTitleSize(0.04);
-   h.DrawClone();
+   h->SetTitle("");
+   h->GetXaxis()->SetTitle("m_{#mu#mu} (GeV)"); h->GetXaxis()->SetTitleSize(0.04);
+   h->GetYaxis()->SetTitle("N_{Events}"); h->GetYaxis()->SetTitleSize(0.04);
+   h->DrawClone();
 
-   TLatex label;
-   label.SetNDC(true);
+   TLatex label; label.SetNDC(true);
    label.DrawLatex(0.175, 0.740, "#eta");
-   label.DrawLatex(0.205, 0.785, "#rho,#omega");
-   label.DrawLatex(0.270, 0.750, "#phi");
+   label.DrawLatex(0.205, 0.775, "#rho,#omega");
+   label.DrawLatex(0.270, 0.740, "#phi");
    label.DrawLatex(0.400, 0.800, "J/#psi");
-   label.DrawLatex(0.415, 0.680, "#psi'");
-   label.DrawLatex(0.485, 0.760, "Y(1,2,3S)");
-   label.DrawLatex(0.755, 0.620, "Z");
-   label.DrawLatex(0.170, 0.350, "#bf{CMS Open Data}");
-   label.DrawLatex(0.170, 0.275, "#bf{#sqrt{s} = 7 TeV}");
-   label.DrawLatex(0.170, 0.200, "#bf{L_{int} = 2.4 fb^{-1}}");
-   label.SetTextSize(0.032);
-   label.DrawLatex(0.10, 0.920, "Run2011A Double Muon Dataset (DOI: 10.7483/OPENDATA.CMS.RZ34.QR6N)");
+   label.DrawLatex(0.415, 0.670, "#psi'");
+   label.DrawLatex(0.485, 0.700, "Y(1,2,3S)");
+   label.DrawLatex(0.755, 0.680, "Z");
+   label.SetTextSize(0.040); label.DrawLatex(0.100, 0.920, "#bf{CMS Open Data}");
+   label.SetTextSize(0.030); label.DrawLatex(0.630, 0.920, "#sqrt{s} = 8 TeV, L_{int} = 11.6 fb^{-1}");
+
+   c->SaveAs("dimuon_spectrum.pdf");
```

Comment:
Add a comment about the prescales in order not to have people commenting about the bump at 30 GeV :)

Reply:
just for the beginners. Let's call it "Trigger effect"